ebml/EbmlElement.h

```diff
@@ -425,11 +425,12 @@ class EBML_DLL_API EbmlElement {
     static bool WriteSkipDefault(const EbmlElement &elt) {
       if (elt.IsDefaultValue())
         return false;
-      return true;
+      return elt.ElementSpec().GetVersions().minver != EbmlDocVersion::ANY_VERSION;
     }
 
-    static bool WriteAll(const EbmlElement &) {
-      return true;
+    // write all elements except deprecated ones
+    static bool WriteAll(const EbmlElement & elt) {
+      return elt.ElementSpec().GetVersions().minver != EbmlDocVersion::ANY_VERSION;
```

Comment:
I don't understand this logic. How is "element is deprecated" the same as "element has a minimum version set"? Don't have elements such as LanguageIETF a minimum version 'cause they were introduced in much later revisions of Matroska?

Reply:
"deprecated" elements (that have minver and maxver set to 0) are marked with a minver of ANY_VERSION:
```
    // the minimum DocType version this element is allowed in
    // ANY_VERSION if the element is never supported
    const num_version minver;
```

Maybe we can add a helper method in the `EbmlDocVersion` class that would hide how it's handeld internally.

Reply:
> Maybe we can add a helper method in the `EbmlDocVersion` class that would hide how it's handeld internally.

Yes, please. That's what I asked for in #207 as well after thinking about it for a bit.